courses/templates/courses/course_list.html

```diff
@@ -1,10 +1,25 @@
 {% extends 'base.html' %}
 
 {% block content %}
-  <h2>Courses</h2>
-  <ul>
+<div class="container-fluid px-4">
+  <h2 class="my-5 text-center">Courses</h2>
+  <div class="row row-cols-1 row-cols-md-2 row-cols-xl-3 g-4">
     {% for course in courses %}
-      <li><a href="{% url 'course' course.slug %}">{{ course.title }}</a></li>
+      <div class="col mb-4">
+        <div class="card h-100 shadow hover-effect">
+          <img src="https://via.placeholder.com/400x250/1E448B/FFF?text={{ course.title|urlencode }}" class="card-img-top" alt="{{ course.title }}" decoding="async">
```

Comment:
What's this website? I'm wondering if it's a good idea to rely on something external

Reply:
I want the course card  to have images like those images [images](https://datatalks.club/images/posts/2024-04-11-guide-to-free-online-courses-at-datatalks-club/image5.png) so I came up with 3 approaches
1. add ImageField to the course model and use it on the course card but this approach needs a lot of work
    - migrations and expected problems that may happen because of courses that exist and don't have an image
    -  a lot of configurations are needed for Django so that Django can serve images correctly on deployment 
2. create  `static/imgs` folder and add those images  by hand and on templates and this requires
    - we should name images using the course name or course slug so that we can add them easily on templates
    - every future course we need to add an image for it on the `static/imgs` folder

3. the third approach and the easiest one use third-party websites that create placeholder images that can be added to a course card

Reply:
Let's do this in a separate PR please. I'd do it via uploading the image to S3 but it's a bigger change for now

Reply:
here is what the last commit introduce
- move all CSS from template to CSS files
- add a default image for the course card so there is no need to use a website that creates placeholder images anymore

the final preview of the website

https://github.com/user-attachments/assets/b24b5080-19f6-4fad-b928-cc4294c41756